src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Date.L.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    public static partial class Utf8Formatter
+    {
+        // Rfc1123 - lowercase
+        //
+        //   01234567890123456789012345678
+        //   -----------------------------
+        //   tue, 03 jan 2017 08:08:05 gmt
+        //
+        private static bool TryFormatDateTimeL(DateTime value, Span<byte> buffer, out int bytesWritten)
+        {
+            bytesWritten = 29;
+            if (buffer.Length < bytesWritten)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            ref byte utf8Bytes = ref buffer.DangerousGetPinnableReference();
+
+            byte[] dayAbbrev = DayAbbreviationsLowercase[(int)value.DayOfWeek];
+            Unsafe.Add(ref utf8Bytes, 0) = dayAbbrev[0];
+            Unsafe.Add(ref utf8Bytes, 1) = dayAbbrev[1];
```

Comment:
Is there a good reason why we are using Unsafe code here? The JIT should be able to eliminate the bounds checks here - because of it should be able to see that Span is at least 29 bytes given the precondition above. If the JIT does not do this optimization, we should get it fixed.

Reply:
It was that way from the start when it was added to CoreFxLab. It was likely a combination of following the pattern used by the variable-lengthed formatters and the fact that we needed the pinnable reference anyway to call the `WriteDigits` helpers. @shiftylogic - if you're still following this, was there a JIT-related reason for using Unsafe.Add here?

Reply:
Switching to the Span indexer on this does seem to slow it down 1-2% (fast Span.)